src/test/clojure/clara/performance/test_compilation.clj

```diff
@@ -0,0 +1,154 @@
+    (println (str \newline "Running Performance tests for:"))
+    (println description)
+    (println "==========================================")
+    (println (str "Mean: " mean "ms"))
+    (println (str "Standard Deviation : " std "ms" \newline))
+    (is (mean-assertion mean))))
+
+(def base-production
+  {:ns-name (symbol (str *ns*))})
+
+(defn generate-filter-productions
+  [seed-syms]
+  (into {}
+        (for [seed-sym seed-syms
+              :let [next-fact (symbol (str seed-sym "prime"))
+                    production (assoc base-production
+                                 :lhs [{:type (keyword seed-sym)
+                                        :constraints ['(= this ?binding) `(filter-fn ~'this)]}]
+                                 :rhs `(r/insert! (with-meta ~(set (repeatedly 10 #(rand-nth (range 100))))
+                                                             {:type ~(keyword next-fact)
+                                                              :val ~'?binding})))]]
+          [next-fact production])))
+
+(defn generate-compose-productions
+  [seed-syms]
+  (let [template (fn template
+                   ([l] (template l l))
+                   ([l r] (let [next-fact (symbol (str l r "prime"))
+                                production (assoc base-production
+                                             :lhs [{:type (keyword l)
+                                                    :constraints ['(= this ?binding-l)]}
+                                                   {:type (keyword r)
+                                                    :constraints ['(= ?binding-l this)]}]
+                                             :rhs `(r/insert! (with-meta ~(set (repeatedly 10 #(rand-nth (range 100))))
+                                                                         {:type ~(keyword next-fact)})))]
+                            [next-fact production])))]
+    (into {}
+          (for [combo (partition-all 2 (shuffle seed-syms))]
+            (apply template combo)))))
+
+(defn generate-collection-productions
+  [seed-syms]
+  (into {}
+        (for [seed-sym seed-syms
+              :let [next-fact (symbol (str seed-sym "prime"))
+                    production (assoc base-production
+                                 :lhs [{:accumulator '(clara.rules.accumulators/all)
+                                        :from {:type (keyword seed-sym),
+                                               :constraints [`(filter-fn ~'this)]}
+                                        :result-binding :?binding}]
+                                 :rhs `(r/insert! (with-meta ~'?binding
+                                                             {:type ~(keyword next-fact)})))]]
+          [next-fact production])))
+
+(defn generate-queries
+  [seed-syms]
+  (into {}
+        (for [seed-sym seed-syms
+              :let [production (assoc base-production
+                                 :lhs [{:type [seed-sym]
```

Comment:
Why is the `:type` wrapped in a vector here? I think it should be `:type seed-sym`.

Reply:
actually, `(keyword seed-sym)`